openeo/rest/auth/oidc.py

```diff
@@ -683,9 +688,14 @@ def _get_verification_info(self, request_refresh_token: bool = False) -> Verific
     def get_tokens(self, request_refresh_token: bool = False) -> AccessTokenResult:
         # Get verification url and user code
         verification_info = self._get_verification_info(request_refresh_token=request_refresh_token)
-        self._display("To authenticate: visit {u} and enter the user code {c!r}.".format(
-            u=verification_info.verification_uri, c=verification_info.user_code)
-        )
+        if verification_info.verification_uri_complete:
+            self._display(
+                f"To authenticate: visit {verification_info.verification_uri_complete}."
```

Comment:
I tried this with a real flow and the point just after the URL is a bit tricky: maybe some tools/context might consider this point to be part of the URL, which will break the authentication. I think we should at least put a space between the URL and "."

Reply:
Good catch!